pyiron_base/job/generic.py

```diff
@@ -1754,6 +1757,15 @@ def _reload_update_master(self, project, master_id):
             self._logger.info("busy master: {} {}".format(master_id, self.get_job_id()))
             del self
 
+    @staticmethod
+    def _register_jobtype_name() -> Union[str, None]:
+        """Name of the JobType this class is to be registered with; None for do not register."""
+        return None
```

Comment:
```suggestion
    @classmethod
    def _register_jobtype_name(cls) -> Union[str, None]:
        """Name of the JobType this class is to be registered with; None for do not register."""
        return cls
```

I suppose we want to do that everywhere by default.  Jobs that specifically don't want to end up in the job class dict can still override this method to do nothing.

Reply:
I tried that and right now (without any further change) all subclasses of `GenericJob` end up in the attached `JOB_CLASS_DICT`. Therefore, we seem to already import all classes?! In addition, all subclasses of master jobs would need to reimplement that method with other than `return None`, i.e. invoke the method of the super super class...
However, overwriting with 
```
if cls.__name__ == "NotToBeRegisteredJobType":
   return None
else:
   return cls.__name__ 
```
would survive naive subclassing :)